python3.pkgs: Migrate fetchers to use hash

mweinelt opened the pull request:
when they already rely on SRI hashes.

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mweinelt committed:
python3.pkgs: Migrate fetchers to use hash

when they already rely on SRI hashes.

mweinelt commented:
```
❯ ./maintainers/scripts/rebuild-amount.sh HEAD~1
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
      3 x86_64-linux

mweinelt commented:
@ofborg eval

mweinelt committed:
aws-cli: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
caffeine-ng: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set an explicit format for buildPythonApplication.

mweinelt committed:
catt: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set an explicit format in buildPythonApplication.

mweinelt committed:
changedetection-io: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
checkov: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
csvs-to-sqlite: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set explicit format in buildPythonApplication.

mweinelt committed:
elasticsearch-curator: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
pgadmin: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
python310Packages.decorator: Use SRI hash

mweinelt committed:
flare-floss: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
privacyidea: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
python310Packages.pyuavcan: Use sri hashes and hash

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt commented:
```
❯ nixpkgs-review pr 219276
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs python-updates:refs/nixpkgs-review/0 pull/219276/head:refs/nixpkgs-review/1
From https://github.com/NixOS/nixpkgs
 + f1501ab5af0d...14e02069b026 refs/pull/219276/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/hexa/.cache/nixpkgs-review/pr-219276-3/nixpkgs cc4e1c29d6754a7dd53aeae8541786751c77a8a4
Preparing worktree (detached HEAD cc4e1c29d675)
Updating files: 100% (33797/33797), done.
HEAD is now at cc4e1c29d675 python310Packages.clevercsv: remove superfluous postPatch
$ nix-env --option system x86_64-linux -f /home/hexa/.cache/nixpkgs-review/pr-219276-3/nixpkgs -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 14e02069b02630ee49427f7c9344766a75932edb
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/hexa/.cache/nixpkgs-review/pr-219276-3/nixpkgs -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
Nothing to be built.

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/219276

$ /nix/store/0g2czg46n8r0n1n7asykjwbk3cv3wc0c-nix-2.13.2/bin/nix-shell /home/hexa/.cache/nixpkgs-review/pr-219276-3/shell.nix
```

dotlambda commented on pkgs/applications/virtualization/docker/compose_1.nix:
```diff
@@ -13,7 +13,7 @@ buildPythonApplication rec {
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-TIzZ0h0jdBJ5PRi9MxEASe6a+Nqz/iwhO70HM5WbCbc=";
+    hash = "sha256-TIzZ0h0jdBJ5PRi9MxEASe6a+Nqz/iwhO70HM5WbCbc=";
```
This change doesn't seem related to awsebcli.

mweinelt replied:
It is related, because awsebcli created a function to update the hash and I converted it to overwrite hashes. It is also used to overwrite the hash for docker-compose.

mweinelt commented on pkgs/tools/virtualization/awsebcli/default.nix:
```diff
@@ -2,22 +2,22 @@
 let
   docker_compose = changeVersion (with localPython.pkgs; docker-compose_1.override {
     inherit colorama pyyaml six dockerpty docker jsonschema requests websocket-client paramiko;
-  }).overridePythonAttrs "1.25.5" "1ijhg93zs3lswkljnm0rhww7gdy0g94psvsya2741prz2zcbcbks";
+  }).overridePythonAttrs "1.25.5" "sha256-ei622Bc/30COUF5vfUl6wLd3OIcZVCvp5JoO/Ud6UMY=";
```
Here.

mweinelt committed:
docker-compose_1: Use SRI hash

To allow awsebcli to use its changeversion function to uniformly
overwrite the hash.

mweinelt committed:
awsebcli: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.
